GitUI/CommandsDialogs/FormPush.cs

```diff
@@ -502,15 +502,15 @@ private bool HandlePushOnExit(ref bool isError, FormProcess form)
 
                 if (forcePush)
                 {
-                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force"))
+                    if (!form.ProcessArguments.Contains(" -f ") && !form.ProcessArguments.Contains(" --force") && !form.ProcessArguments.EndsWith(" -f"))
```

Comment:
It is not actual anymore, but answer is: I add -f to end of arguments, so check .Contains(" -f ") will not work because of trailing space.

So this condition should cover all possible cases:

push -f origin
push --force origin
push --force-with-lease origin

push origin ... -f
push origin ... --force
push origin ... --force-with-lease